lib/src/cwrapper.cc

```diff
@@ -76,3 +78,47 @@ uint8_t get_resource(char **result, char *type_name, char *resource_name) {
 
     return EXIT_SUCCESS;
 }
+
+uint8_t set_resource(char **result, char *type_name, char *resource_name, int desired_attributes_c, char **desired_attributes) {
+    std::vector<std::string> data_dirs;
+    auto ral = lib::ral::create(data_dirs);
+    auto opt_type = ral->find_type(std::string(type_name));
+
+    if (!opt_type) {
+        str_to_cstr("Provider not found", result);
+        return EXIT_FAILURE;
+    }
+
+    if (desired_attributes_c < 1) {
+        str_to_cstr("Number of desired attributes must be > 0", result);
+        return EXIT_FAILURE;
+    }
+    
+    std::vector<std::string> av(desired_attributes, desired_attributes + desired_attributes_c);
+
+    auto& type = *opt_type;
+    lib::resource should = type->prov().create(std::string(resource_name));
+
+    for (const auto& arg : av) {
+        auto found = arg.find("=");
+        if (found != std::string::npos) {
+            auto attr = arg.substr(0, found);
+            auto value = type->parse(attr, arg.substr(found+1));
+            if (value) {
+                should[attr] = value.ok();
+            } else {
+                str_to_cstr("Error reading attribute", result);
+                fprintf( stderr, "Failed to read attribute %s, resource_name: %s\n", attr.c_str(), value.err().detail.c_str() );
```

Comment:
@lutter I tried to return the error using `snprintf` but couldn't get it working correctly, was passing a null buffer to first get the size then allocating that size+1 and using that with `sprintf` but it was causing a segfault. Is there a canonical way of doing this?